src/drivers/imx/interrupt.c

```diff
@@ -159,11 +159,10 @@ static void irqstr_disable_hw(void)
  */
```

Comment:
@paulstelian97 PR looks good to me. Tested on i.MX8QXP works fine. 

Care to enhance to commit message to explain why the bounds check are no longer correct for i.MX8M. Just a short summary for someone who is not familiar with inner details of irqsteer.